hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreScanner.java

```diff
@@ -635,8 +635,14 @@ public boolean next(List<Cell> outResult, ScannerContext scannerContext) throws
             scannerContext.incrementBatchProgress(1);
 
             if (matcher.isUserScan() && totalBytesRead > maxRowSize) {
-              throw new RowTooBigException(
-                  "Max row size allowed: " + maxRowSize + ", but the row is bigger than that.");
+              LOG.warn("Max row size allowed: {}, but the row is bigger than that, the row info: "
+                  + "{}, already have process row cells = {}, it belong to region = {}",
+                maxRowSize, CellUtil.toString(cell, true), outResult.size(),
+                store.getHRegion().getRegionInfo().getRegionNameAsString());
+              throw new RowTooBigException("Max row size allowed: " + maxRowSize
+                + ", but the row is bigger than that, the row info: " + CellUtil
+                .toString(cell, true) + ", it belong to region = {}" + store.getHRegion()
```

Comment:
I think the variable substitution (`{}`) here won't work. Didn't we had a better way for this in exceptions?

Reply:
Oh yes, it would not work here. It works only for LOG. Looks like recent commit change @binlijin

Reply:
ok, fix it.